### ISSUE TITLE
Deterministic sys call / memory fuzzing

### ISSUE THREAD

**Issue by ronaldtse** [riboseinc/retrace#156]
Currently we can use retrace to discover bugs through memory fuzzing (such as riboseinc/rnp#255), but it would be really great if there is an option to fuzz with a given seed so it's deterministically reproducible.

This way projects can incorporate retrace in CI to ensure previous bugs do not reoccur. 

cc @riataman @catap

**Comment by riataman** [comment]
Yeah, this makes sense. We need to include our own random generator otherwise we risk the traced program also calling srand()/srandom() before/after we do.

**Comment by jjr840430** [comment]
@ronaldtse 
Please review the following configuration.
memoryfuzzing,[fuzzing rate],[seed value]

**Comment by catap** [comment]
@jjr840430 and I think we should add to log current seed value to easy reproduce the bug

**Comment by jjr840430** [comment]
@catap If we use fixed seed value, then is it need to get seed value from log file?

**Comment by ronaldtse** [comment]
@jjr840430 the configuration line looks good. As @catap we should also print out the current seed in the trace log. While the seed is already in the config, it is clearer in the trace that it was using a deterministic random.

**Comment by jjr840430** [comment]
Right, I will do so

**Comment by ronaldtse** [comment]
Shipped: deterministic fuzzing via `fuzzing_seed` (config) and `RETRACE_FUZZ_SEED` (env) — same seed, same failure pattern; documented in docs/configuration.md and cookbook 11.
